runtime/lua/lsp/plugin.lua

```diff
@@ -0,0 +1,137 @@
+local Client = require('lsp.client')
```

Comment:
What does "plugin" mean here? This seems to just be methods that are part of the public interface. Perhaps this file should just be `runtime/lua/lsp.lua` (or rather `runtime/lua/vim/lsp.lua` , we should scope everything under `vim` to avoid namespace collision).